tests/test_mpls.py

```diff
@@ -0,0 +1,1049 @@
+    def set_admin_status(self, interface, status):
+        self.cdb.update_entry("PORT", interface, {"admin_status": status})
+
+    def create_l3_intf(self, interface):
+        self.cdb.create_entry("INTERFACE", interface, {"NULL": "NULL"})
+
+    def remove_l3_intf(self, interface):
+        self.cdb.delete_entry("INTERFACE", interface)
+
+    def create_mpls_intf(self, interface):
+        self.cdb.create_entry("INTERFACE", interface, {"mpls": "enable"})
+
+    def remove_mpls_intf(self, interface):
+        self.cdb.delete_entry("INTERFACE", interface)
+
+    def add_ip_address(self, interface, ip):
+        self.cdb.create_entry("INTERFACE", interface + "|" + ip, {"NULL": "NULL"})
+
+    def remove_ip_address(self, interface, ip):
+        self.cdb.delete_entry("INTERFACE", interface + "|" + ip)
+
+    def create_route_entry(self, key, pairs):
+        tbl = swsscommon.ProducerStateTable(self.pdb.db_connection, "ROUTE_TABLE")
+        fvs = swsscommon.FieldValuePairs(list(pairs.items()))
+        tbl.set(key, fvs)
+
+    def remove_route_entry(self, key):
+        tbl = swsscommon.ProducerStateTable(self.pdb.db_connection, "ROUTE_TABLE")
+        tbl._del(key)
+
+    def check_route_entries(self, destinations):
+        def _access_function():
+            route_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+            route_destinations = [json.loads(route_entry)["dest"]
+                                  for route_entry in route_entries]
+            return (all(destination in route_destinations for destination in destinations), None)
+
+        wait_for_result(_access_function)
+
+    def get_asic_db_key(self, destination):
+        route_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+        for route_entry in route_entries:
+            if json.loads(route_entry)["dest"] == destination:
+                return route_entry
+        return None
+
+    def check_route_entries_nexthop(self, destinations, vrf_oids, nexthops):
+        def _access_function_nexthop():
+            nexthop_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP")
+            nexthop_oids = dict([(self.adb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP", key)["SAI_NEXT_HOP_ATTR_IP"], key)
+                                 for key in nexthop_entries])
+            return (all(nexthop in nexthop_oids for nexthop in nexthops), nexthop_oids)
+
+        status, nexthop_oids = wait_for_result(_access_function_nexthop)
+
+        def _access_function_route_nexthop():
+            route_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+            route_destination_nexthop = dict([((json.loads(route_entry)["dest"], json.loads(route_entry)["vr"]), 
+                                               self.adb.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY", route_entry).get("SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID"))
+                                               for route_entry in route_entries])
```

Comment:
Can we check label stack values and outseg types in mpls nexthop?

Reply:
@kperumalbfn Done.  See latest checking with check_nexthop().